frame/crowdloan-bonus/src/lib.rs

```diff
@@ -142,7 +144,11 @@ pub mod pallet {
 
 			ensure!(pot_balance_value > 0 && token_supply_value > 0, Error::<T>::EmptyPot);
 
-			let to_payout = pot_balance_value.mul(amount).div(token_supply_value);
+			let to_payout = pot_balance_value
+				.checked_mul(amount)
+				.and_then(|x| x.checked_div(token_supply_value))
+				.ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+
```

Comment:
ArithmeticError can be division by zero also. Could we use `safe_div`?

Reply:
Of course!